include/bluetooth/conn.h

```diff
@@ -711,8 +711,78 @@ struct bt_conn_oob_info {
 	};
 };
 
+#if defined(CONFIG_BT_SMP_PAIRING_ALLOW)
+/* Info structure for generic_pairing_query */
+struct bt_conn_pairing_auth_req_info {
+	/** AuthReq, Core Spec. Vol 3, Part H, 3.5.1, Fig. 3.3 */
+	u8_t bonding_flags:2,
+	     mitm:1,
+	     sc:1,
+	     keypress:1,
+	     ct2:1,
+	     rfu:2;
```

Comment:
This is unportable since you're assigning to it the raw u8_t value from the protocol. The C standard doesn't define how bitfields are ordered in memory (it's left to be implementation-defined). Usually little- and big-endian architectures do this differently.

We've generally steered away from bitfields when it comes to protocol values, and only used them for internal variables where compactness was desired. Instead we use helper macros to encode and decode to the protocol format.